src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
+            {
+                Exception savedEx = (Exception)rtRequest.Properties[SavedExceptionLookupKey];
+                if (savedEx != null)
+                {
+                    throw savedEx;
```

Comment:
This is wrapped in an HttpRequestException here:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs#L577